doc/edgehub-api.md

```diff
@@ -0,0 +1,434 @@
+
+### Telemetry messages
+In case of devices, a telemetry message can be sent to Edge Hub 1.2 if the MQTT broker is turned on using the following topics:
+
+If the client is a device:
+```
+$iothub/{device_id}/messages/events
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/messages/events
+```
+Other parameters can be encoded into the topic similarly to the legacy telemetry topics.
+### Getting Full Twin
+If a client wants to be able to receive the result of its twin request, then it needs to subsribe to the following topic:
+
+If the client is a device:
+```
+$iothub/{device_id}/twin/res/#
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/twin/res/#
+```
+After the subscription, the client can request a twin by publishing an empty message on the topic:
+
+If the client is a device:
+```
+$iothub/{device_id}/twin/get/?$rid={request_id}
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/twin/get/?$rid={request_id}
+```
+Edge Hub is going to send the twin publishing on a topic similar to the following:
+```
+$iothub/some_device/some_module/twin/res/200/?$rid=1234
+```
+### Sending Reported Property Updates
+For sending propery updates, the client first needs to subscribe to the twin response topic just like when it pulls the full twin. After the subscription, the client can publish its update on the following topic:
+
+If the client is a device:
+```
+$iothub/{device_id}/twin/reported/?$rid={request_id}
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/twin/reported/?$rid={request_id}
+```
+### Receiving Desired Property Updates
+The topic to subscribe for desired properties,
+
+If the client is a device:
+```
+$iothub/{device_id}/twin/desired/#
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/twin/desired/#
+```
+### Receiving Cloud-to-Device messages
+In order to receive C2D messages, the client needs to subscribe to the following topic:
+```
+$iothub/{device_id}/messages/c2d/post/#
+```
+### Receiving Direct-Method calls
+To be able to receive direct method calls, a client needs to subscribe to:
+
+If the client is a device:
+```
+$iothub/{device_id}/methods/post/#
+```
+If the client is a module:
+```
+$iothub/{device_id}/{module_id}/methods/post/#
+```
+
+Edge Hub is going to send a direct method call similarly to the following:
+```
+$iothub/some_device/some_module/methods/post/some_method/?$rid=1234
+```
+In response the client needs to publish on the following topic:
+
+```
+$iothub/some_device/some_module/methods/res/200/?$rid=1234
+```
+## Edge Hub HTTPS Endpoint
```

Comment:
Can we add documentation for the HTTPS endpoints like this - https://github.com/Azure/iotedge/blob/master/edgelet/edgelet-http-workload/doc/list_modules.md